core/src/mindustry/input/InputHandler.java

```diff
@@ -109,9 +109,9 @@ public static void clearItems(Building build){
 
     @Remote(called = Loc.server, unreliable = true)
     public static void transferItemTo(@Nullable Unit unit, Item item, int amount, float x, float y, Building build){
-        if(build == null || build.items == null) return;
+        if(build == null || build.items == null || unit == null || unit.stack.amount <= 0) return;
```

Comment:
Buildings should still handle items even if the unit is null. Not doing this will cause opportunities for desync.